### DISCUSSION TITLE
How to setup pyright-langserver

### DISCUSSION THREAD

**Discussion by littledian** [microsoft/pyright#4495] [Q&A]
I want to make a website, that can online edit python code, Now I succesuufy run it, by using monaco-editor and vscode-ws-jsonrpc for code assistant, but a little pity is that when I use pyright as langservice (just node  langserver.index.js), It can not recommend the python packages that I have installed, also can not recommend the method of the package I imported, so how to do it?

**Comment by erictraut** [comment]
You may be interested in https://vscode.dev/. It provides an online editing experience based on the monaco editor and incorporates pyright.

**Reply by littledian** [reply]
Thank you for reply, I have seen it, and I think it may weight for us, we just nned a light code editor just only collect code to server

**Reply by hareeshmuddala** [reply]
Hey @littledian  , Actually i am very new to this lsp thing,i am trying to implement the same i used monaco editor for frontend and now i want auto suggestion for that i want to use lsp but i am unable find  any resources could you please point me to some resources , as per my understanding i need to up serever and connect app through websocket and should use jsonrpc protocol am i right ?
